### DISCUSSION TITLE
[Bug]: Processing update index queue via Symfony messenger workers [Portal engine]

### DISCUSSION THREAD

**Discussion by yamenalahlas** [pimcore/pimcore#13223] [Q&A / Support] [answered]
### Expected behavior

Processing update index queue via Symfony messenger workers can be successfully executed.

### Actual behavior

Processing update index queue via Symfony messenger workers cannot be successfully executed.

Error:
The receiver "pimcore_portal_engine" does not exist. Valid receivers are: pimcore_core, pimcore_maintenance, pimcore_image_optimize, pimcore_index_queues, pimcore_data_import, pimcore_datahub_fileexport, portal-engine-async.  

### Steps to reproduce

Add Portal engin configs:
`pimcore_portal_engine:
    messenger_queue_processing:
      activated: true`
run Command `$ bin/console messenger:consume pimcore_portal_engine pimcore_index_queues  --time-limit=300`
Error:
`The receiver "pimcore_portal_engine" does not exist. Valid receivers are: pimcore_core, pimcore_maintenance, pimcore_image_optimize, pimcore_index_queues, pimcore_data_import, pimcore_datahub_fileexport, portal-engine-async. `

when I run the Command with  portal-engine-async instead of pimcore_portal_engine like this: `$ bin/console messenger:consume portal-engine-async pimcore_index_queues  --time-limit=300 `
The Indexes in Portal will be not updated

**Comment by mcop1** [comment]
@yamenalahlas Do you get any error messages? What if you add "-vv" option to the messenger:consume command?

**Comment by yamenalahlas** [comment]
yes, it throws some exceptions,
11:45:25 INFO      [php] User Deprecated: Specifying types in urls has been deprecated
12:00:24 INFO      [php] Deprecated: strip_tags(): Passing null to parameter #1 ($string) of type string is deprecated
12:03:44 INFO      [php] Deprecated: Return type of 
Pimcore\Bundle\PortalEngineBundle\Model\BasicJsonModel::jsonSerialize() should either be compatible with JsonSerializable::jsonSerialize(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice
12:03:44 INFO      [php] User Deprecated: Since symfony/http-foundation 5.3: "Symfony\Component\HttpFoundation\RequestStack::getMasterRequest()" is deprecated, use "getMainRequest()" instead.

12:03:44 INFO      [php] User Deprecated: Since pimcore/pimcore 10.5: Using $force=false on Pimcore\Model\DataObject\AbstractObject::getById is deprecated, please use array-syntax [force=>true] instead.

**Comment by mcop1** [comment]
The deprecations aren´t 100% ideal, but they shouldn´t be a problem. I moved your issue to the discussions, maybe someone there had similiar problems and can help.

**Comment by Maksonis** [comment]
You should use portal-engine-async instead of pimcore_portal_engine. Check file: vendor/pimcore/portal-engine/src/Resources/config/pimcore/messenger.yml

![image](https://user-images.githubusercontent.com/17277953/192796550-849b127e-6294-45bf-8305-db6d064e0e8e.png)

I think this should be fixed in documentation.

**Comment by yamenalahlas** [comment, marked as answer]
Okay, today I finally fixed it. First I had to update the indices structure once and then to update the indices then will work with symfony messenger. @mcop1 
There is also the docs one mistake
`portal_engine:
    messenger_queue_processing:
        activated: true`
It has to be 
`pimcore_portal_engine:
    messenger_queue_processing:
        activated: true`

At this site: https://pimcore.com/docs/portal-engine/current/Installation/Installation.html

This command has to be changed from:
`php bin/console messenger:consume pimcore_portal_engine pimcore_index_queues`
to:
`php bin/console messenger:consume portal-engine-async pimcore_index_queues`
At this site: https://pimcore.com/docs/portal-engine/current/Administration_of_Portals/Background_Processes.html
Suggestion:
It is very helpful if gives a hint: when using Symfony Messenger for updating indices, first you need to create the indices structure and update the indices.

**Comment by mcop1** [comment]
@yamenalahlas 
Sounds great and thanks for the info on the docs. Do you want to create a pr in the portal-engine repository to fix it? Otherwise I will change it next week.
